cli/pcluster/config_sanity.py

```diff
@@ -532,16 +534,59 @@ def validate(self, resource_type, resource_value):  # noqa: C901 FIXME
             ]:
                 self.__fail(resource_type, "Region %s is not supported with batch scheduler" % self.region)
 
+            # Check spot bid percentage
+            if "SpotPrice" in resource_value:
+                spot_price = int(resource_value["SpotPrice"])
+                if spot_price > 100 or spot_price < 0:
+                    self.__fail(resource_type, "Spot bid percentage needs to be between 0 and 100")
+
+            # Check sanity on desired, min and max vcpus
+            if (
+                "DesiredSize" not in resource_value
+                or "MinSize" not in resource_value
+                or "DesiredSize" not in resource_value
+            ):
+                # this should never occur
+                self.__fail(resource_type, "Both min, desired and max vcpus parameters must be set")
```

Comment:
Is this true? can't I specify only some of them?

Also you are checking twice for DesiredSize

Reply:
Looking at `__init_batch_parameters()`, there's no way these values would not be set. There is a chance they're set to a string type, but this doesn't validate that.